tests/agent_server/test_server_agent.py

```diff
@@ -3430,3 +3430,63 @@ async def test_agentinstance_stops_deploying_when_stopped(
     await resource_container.wait_for_done_with_waiters(
         client, environment, version, wait_for_this_amount_of_resources_in_done=2
     )
+
+
+@pytest.mark.asyncio
+async def test_set_fact_in_handler(server, client, environment, agent, clienthelper, resource_container, no_agent_backoff):
+    """
+        Test whether facts set in the handler via the ctx.set_fact() method arrive on the server.
+    """
+
+    def get_resources(version: str, params: List[data.Parameter]) -> List[Dict[str, Any]]:
+        return [
+            {
+                "key": param.name,
+                "value": param.value,
+                "metadata": param.metadata,
+                "id": f"{param.resource_id},v={version}",
+                "send_event": False,
+                "purged": False,
+                "purge_on_delete": False,
+                "requires": [],
+            }
+            for param in params
+        ]
+
+    def compare_params(actual_params: List[data.Parameter], expected_params: List[data.Parameter]) -> None:
+        actual_params = sorted(actual_params, key=lambda p: p.name)
+        expected_params = sorted(expected_params, key=lambda p: p.name)
+        assert len(expected_params) == len(actual_params)
+        for i in range(len(expected_params)):
+            for attr_name in ["name", "value", "environment", "resource_id", "source", "metadata"]:
+                assert expected_params[i].__getattribute__(attr_name) == actual_params[i].__getattribute__(attr_name)
```

Comment:
I think the proper way of doing that is with

```suggestion
                assert getattr(expected_params[i], attr_name) == getattr(actual_params[i], attr_name)
```